src/udev/udev-builtin-net_id.c

```diff
@@ -161,6 +167,63 @@ static struct udev_device *skip_virtio(struct udev_device *dev) {
         return parent;
 }
 
+static int get_virtfn_info(struct udev_device *dev, struct netnames *names, struct virtfn_info *vf_info) {
+        struct udev *udev;
+        const char *physfn_link_file;
+        _cleanup_free_ char *physfn_pci_syspath = NULL;
+        _cleanup_free_ char *virtfn_pci_syspath = NULL;
+        struct dirent *dent;
+        _cleanup_closedir_ DIR *dir = NULL;
+        struct virtfn_info vf_info_local;
+        int rc = -ENOENT;
+
+        udev = udev_device_get_udev(names->pcidev);
+        if (!udev)
+                goto out;
+        /* Check if this is a virtual function. */
+        physfn_link_file = strjoina(udev_device_get_syspath(names->pcidev), "/physfn");
+        if (chase_symlinks(physfn_link_file, NULL, 0, &physfn_pci_syspath) <= 0)
+                goto out;
```

Comment:
`return rc` should suffice here.

Reply:
that said, I think it would be nicer to propagate the error chase_symlinks() returned here.